File: backend/src/main/java/com/unibuc/find_my_path/model/TestAttempt.java
package com.unibuc.find_my_path.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class TestAttempt {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long testAttemptId;

    @Column(nullable = false)
    private LocalDateTime attemptDate;

    private Integer testRating;
    private Integer experienceRating;

    @Column(nullable = false)
    private Boolean isCompleted;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private FindMyPathUser user;

    @OneToOne(mappedBy = "testAttempt", cascade = CascadeType.ALL)
    private Result result;

    @ManyToMany
    @JoinTable(
            name = "test_attempt_answers",
            joinColumns = @JoinColumn(name = "test_attempt_id"),
            inverseJoinColumns = @JoinColumn(name = "answer_id")
    )
    private List<Answer> answerList;
}

File: frontend/lib/models/test_attempt.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'test_attempt.freezed.dart';
part 'test_attempt.g.dart';

@freezed
class TestAttempt with _$TestAttempt {
  factory TestAttempt({
    @JsonKey(name: 'test_attempt_id') required int testAttemptId,
    @JsonKey(name: 'user_id') required String userId,
    @JsonKey(name: 'attempt_date') required DateTime? attemptDate,
    @JsonKey(name: 'test_rating') required int? testRating,
    @JsonKey(name: 'experience_rating') required int? experienceRating,
    @JsonKey(name: 'is_completed') required bool isCompleted,
  }) = _TestAttempt;

  factory TestAttempt.fromJson(Map<String, dynamic> json) =>
      _$TestAttemptFromJson(json);
}
